test/test_unistd.rs

```diff
@@ -11,16 +11,17 @@ use std::io::Write;
 use std::os::unix::prelude::*;
 use tempfile::tempfile;
 use tempdir::TempDir;
-use libc::off_t;
-use std::process::exit;
+use libc::{_exit, off_t};
 
 #[test]
 fn test_fork_and_waitpid() {
     #[allow(unused_variables)]
     let m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
-    let pid = fork();
+
+    // Safe: Child only calls `_exit`, which is signal-safe
+    let pid = unsafe { fork() };
     match pid {
```

Comment:
Let's get rid of the `pid` variable completely here for consistency with other tests and since it isn't used anywhere else.